### ISSUE TITLE
Disabled time not checking for PM timing

### ISSUE THREAD

**Issue by Benana89** [mengxiong10/vue2-datepicker#664]
`<date-picker
                v-model="newDate"
                :value-type="YYYY-MM-DD hh:mm A"
                :format="hh:mm A"
                :type="time"
                :default-value="defaultDate"
                :disabled-time="disableTime"></date-picker>`

I set the var disableTime to Tue Nov 30 2021 15:39:00 GMT+0800 and my newDate to 2021-11-30 00:00 AM. All the selection for timepicker dropdown is being disabled, and the date param for the disableTime, it only return me date time from 1 -12 AM, even i set the disabled to be 15:39PM, How can i check for PM timing?

**Comment by mengxiong10** [comment]
which version do you use? And can you should you code with more detail?

**Comment by Benana89** [comment]
Vue2-datepicker version: 3.9.2
Vue version: 2.6.11

time reproduce: 3:00 PM
**Steps to reproduce**

Set `:format="hh:mm A"` to display as 12 hr 
Set the v-model to 12:00 AM
Set `:disabled-time` to disableTime function and disableTimeValue set to 1:00 PM
```
disableTime(date: Date): boolean {
    if (this.disableTimeValue) {
      
      return date.getTime() < this.disableTimeValue.getTime();
    }
    return false;
  };
```


**Expected behavior**

AM selection is disabled
PM selection can be selected for time after 1:00PM

**Actual behavior**

All the time selection in Time dropdown is disabled
<img width="316" alt="Screenshot 2021-12-02 at 10 13 51 AM" src="https://user-images.githubusercontent.com/48176330/144344943-bd27ef91-d014-4663-8b13-33fcf7717afc.png">

**Comment by mengxiong10** [comment]
You can use v3.10.x.

**Comment by Benana89** [comment]
Updated to v3.10.3

The date param in disableTime return only the range from 00:00 - 12:00 GMT which is not in range for my disable timing 13:00 GMT
<img width="1209" alt="Screenshot 2021-12-02 at 3 10 37 PM" src="https://user-images.githubusercontent.com/48176330/144375329-0dc8c076-7243-4264-87b8-322f15201cf3.png">

**Comment by mengxiong10** [comment]
Thanks for feedback. I have fixed it in 3.10.4
